lib/id3c/db/__init__.py

```diff
@@ -56,37 +56,24 @@ def mint_identifiers(session: DatabaseSession, name: str, n: int) -> Any:
 
     started = datetime.now()
 
-    while len(minted) < n:
-        m = len(minted) + 1
+    minted = session.fetch_all("""
+        select uuid, barcode, identifier_set_id, generated
+        from mint_identifiers(%s, %s)
+        """, (identifier_set.id, n))
```

Comment:
You mentioned this to me but wanted to note here too, does it make sense to wrap this inside a savepoint? Not sure if it matters or not since everything is handled in the Postgres side now but maybe it would protect from some edge cases with the DB connection that aren't immediately apparent.

Reply:
I don't believe the savepoint is needed anymore since the iteration is being handled in PostgreSQL function, and the entire transaction will either succeed or be rolled back if it fails. The previous implementation was using them to remove inserts that failed from from the transaction, but I'm not sure that's actually necessary.

Reply:
Confirmed we shouldn't need savepoints here, and should expect the same behavior because postgres automatically rolls back any changes since the begin statement when there's an exception: [42.13.2.1. Implicit Rollback After Exceptions](https://www.postgresql.org/docs/13/plpgsql-porting.html#PLPGSQL-PORTING-OTHER)

Reply:
Thanks for looking into it @davereinhart